src/librustc_resolve/lib.rs

```diff
@@ -3512,6 +3668,41 @@ fn path_names_to_string(path: &Path, depth: usize) -> String {
     names_to_string(&names[..])
 }
 
+fn candidates_to_suggestion(paths: &Vec<Path>) -> Vec<String> {
+
+    let mut suggestion_vec = Vec::new();
+    let mut path_strings: Vec<_> = paths.into_iter()
+                                        .map(|p| path_names_to_string(&p, 0))
+                                        .collect();
+    // we want consistent results across executions, but candidates are produced
+    // by iterating through a hash map, so make sure they are ordered:
+    path_strings.sort();
+
+    if !paths.is_empty() {
+        if paths.len() == 1 {
```

Comment:
In other cases like this, we typically limit the total number of suggestions we offer to 4 -- if there are more than 4, then we print "and N others" as the final item (so that we list 5 items all in all). We should do the same here.